storage/range_command.go

```diff
@@ -592,11 +592,11 @@ func (r *Range) InternalPushTxn(batch engine.Engine, ms *proto.MVCCStats, args *
 		pusherWins = false
 	} else if reply.PusheeTxn.Priority < priority ||
 		(reply.PusheeTxn.Priority == priority && args.Txn != nil &&
-			args.Txn.Timestamp.Less(reply.PusheeTxn.Timestamp)) {
-		// Pusher wins based on priority; if priorities are equal, order
-		// by lower txn timestamp.
+			bytes.Compare(args.Txn.ID, reply.PusheeTxn.ID) > 0) {
```

Comment:
Hmmm, why do you need the third transaction?

Also why not use timestamp first before using txn-id ?

On Mon, Jun 1, 2015 at 10:10 PM Tobias Schottdorf <notifications@github.com>
wrote:

> In storage/range_command.go
> <https://github.com/cockroachdb/cockroach/pull/1216#discussion_r31488453>:
>
> > @@ -592,11 +592,11 @@ func (r *Range) InternalPushTxn(batch engine.Engine, ms *proto.MVCCStats, args *
> >  		pusherWins = false
> >  	} else if reply.PusheeTxn.Priority < priority ||
> >  		(reply.PusheeTxn.Priority == priority && args.Txn != nil &&
> > -			args.Txn.Timestamp.Less(reply.PusheeTxn.Timestamp)) {
> > -		// Pusher wins based on priority; if priorities are equal, order
> > -		// by lower txn timestamp.
> > +			bytes.Compare(args.Txn.ID, reply.PusheeTxn.ID) > 0) {
>
> If I didn't get something wrong then a very common mechanism for getting
> this is two txns running into a third, and then into each other. So it's
> very deterministic.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cockroachdb/cockroach/pull/1216/files#r31488453>.
>